### PR TITLE
Add flow to ObjectInspector.js, Rep.js, Svg.js,...

### PR DESCRIPTION
**Pull request by arthur801031**
Associated Issue: #1894

### Summary of Changes

Add flow to the following:
- shared/ObjectInspector.js
- shared/Rep.js
- shared/Svg.js
- shared/Menu.js
- Editor/index.js

### Test Plan
Passed 
- `yarn run flow`
- `npm run lint-css -s; npm run lint-js -s`
- `node src/test/node-unit-tests.js --dots`

### CODE REVIEW

File: src/components/shared/Popover.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ class Popover extends Component {
     const left = targetLeft + targetWidth / 2 - width / 5;
     const top = targetBottom;
 
+    // eslint-disable-next-line react/no-did-mount-set-state
```
Comment (arthur801031) [inline src/components/shared/Popover.js]: Fix the following warning:
screen shot 2017-03-07 at 10 34 54 am

File: src/components/Editor/ConditionalPanel.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ const CloseButton = require("../shared/Button/Close");
 require("./ConditionalPanel.css");
 
 function renderConditionalPanel({ condition, closePanel, setBreakpoint }:
-  { condition: boolean, closePanel: Function, setBreakpoint: Function }) {
```
Comment (jasonLaster) [inline src/components/Editor/ConditionalPanel.js]: 👍

File: src/components/shared/Rep.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
+// @flow
 const React = require("react");
 
+import type { ObjectInspectorItemContentsValue } from "../../types";
+
+type RepRender = {
```
Comment (jasonLaster) [inline src/components/shared/Rep.js]: perhaps `RenderRepOptions`

File: src/reducers/sources.js [hunk 4]
```diff
@@ -283,8 +283,8 @@ function getSources(state: OuterState) {
   return state.sources.sources;
 }
 
-function getSourceText(state: OuterState, id: string) {
-  return state.sources.sourcesText.get(id);
+function getSourceText(state: OuterState, id: ?string) {
+  return id ? state.sources.sourcesText.get(id) : undefined;
```
Comment (jasonLaster) [inline src/reducers/sources.js]: hmm, 

```js
if (id) {
  return state.sources.sourcesText.get(id) 
}

File: src/types.js [hunk 5]
```diff
@@ -22,6 +22,18 @@ export type Mode = String | {
...
+    actor: string,
+    class: string,
+    displayClass: string,
+    extensible: boolean,
+    frozen: boolean,
+    ownPropertyLength: number,
+    preview: Object,
+    sealed: boolean,
+    type: string
+};
```
Comment (jasonLaster) [inline src/types.js]: I think i'd prefer having `ObjectInspector.js` or `object-inspector` export the type.

**Pull request merged**